python_modules/libraries/dagster-k8s/dagster_k8s/component.py

```diff
@@ -0,0 +1,62 @@
+from collections.abc import Mapping, Sequence
+from dataclasses import dataclass
+from functools import cached_property
+from typing import Any, Optional, Union
+
+from dagster import AssetExecutionContext, AssetsDefinition, Definitions, multi_asset
+from dagster.components import Component, ComponentLoadContext, Resolvable, ResolvedAssetSpec
+
+from dagster_k8s.pipes import PipesK8sClient, build_pod_body
+
+
+@dataclass
+class PipesK8sSpec(Resolvable):
+    name: str
+    assets: Sequence[ResolvedAssetSpec]
+    image: Optional[str] = None
+    command: Optional[Union[str, Sequence[str]]] = None
+    namespace: Optional[str] = None
+    env: Optional[Mapping[str, str]] = None
+    base_pod_meta: Optional[Mapping[str, Any]] = None
+    base_pod_spec: Optional[Mapping[str, Any]] = None
+
+    def __post_init__(self):
+        # validate that we can build a pod for the given args
+        # i.e. image or base_pod_spec.image
+        build_pod_body(
+            pod_name=self.name,
+            image=self.image,
+            command=self.command,
+            env_vars=self.env or {},
+            base_pod_meta=self.base_pod_meta,
+            base_pod_spec=self.base_pod_spec,
+        )
+
+
+@dataclass
+class PipesK8sCollectionComponent(Component, Resolvable):
+    """Component that creates assets backed by kubernetes pod execution via Dagster Pipes."""
+
+    specs: Sequence[PipesK8sSpec]
```

Comment:
Like the subprocess component, it just takes a list of descriptors for what to build. 

You could imagine a version that takes a folder path (or just crawls relative to the yaml file location) expecting each descriptor in a separate yaml file likely next to its Dockerfile. I am skeptical that that would be compatible with the wide range of setups for container definition that users could have.